### DISCUSSION TITLE
Performance tunning section is missing for the docs

### DISCUSSION THREAD

**Discussion by FarhadF** [openobserve/openobserve#1253] [General]
### Describe the problem that you experienced

I dont know how to optimize performance on openobserve. Currently ingestion stats looking like this for ~24 hours of data:

![image](https://github.com/openobserve/openobserve/assets/17374492/20e8b681-6700-43ad-93f8-7699e93110f4)

Query stats with no fields for last 15 minutes:
![image](https://github.com/openobserve/openobserve/assets/17374492/b36f14e2-0d00-46cd-a4f5-28566a44c500)

Added partitions on application name, and log_level.

Query using the partition:
```fields_application='xxx' AND level_name='info'```

![image](https://github.com/openobserve/openobserve/assets/17374492/f2c8229e-9877-4aef-be36-152fe0c75975)

I have 6 queriers with cache enabled  + 2 routers and using s3 as storage backend. Is this performance expected? How can I improve the search performance?

**Comment by prabhatsharma** [comment]
This is super low performance compared to what we see in general. Can you help us understand the capacity of each querier and node (vCPUs) and available RAM.

Generally, you would get the performance of 700 MB to 3.5 GB per second per CPU core based on multiple factors. You could increase this to a great degree by partitioning further. In your case you could partition by `fields_application` and `level_name` to get even more performance.

We are making some changes to the core engine to make it even more performant and will update the docs soon in this area. Thanks for pointing this out.

**Comment by FarhadF** [comment]
There are no CPU/Memory requests/limits on the openobserve pods. The nodes in the cluster are aws ec2 instance m5.2xlarge/m5.4xlarge.

About partitioning I have already the partitions setup on fields_application and level_name and I can verify that they are in place by s3 folder structure.

**Comment by prabhatsharma** [comment]
I don't think we have enough information around what is going on that is causing such low performance. Also, why don't you use m6i instances instead of m5 instances. m7g instances would be even better. See if that helps. I would also request you join our [slack channel](https://join.slack.com/t/zincobserve/shared_invite/zt-11r96hv2b-UwxUILuSJ1duzl_6mhJwVg). It's a lot easier to communicate and troubleshoot issues there.

**Comment by prabhatsharma** [comment]
Can you try removing the partition. If there are too many files in the partition then it may cause slowness. Can you also share OpenObserve logs and version number.

**Comment by hengfeiyang** [comment]
@FarhadF Can you give some logs of OpenObserve, it should print some details, like how many files

**Comment by prabhatsharma** [comment]
We have a performance optimization section now - https://openobserve.ai/docs/performance
